files/scripts/workspaces_launch_test.py

```diff
@@ -0,0 +1,225 @@
+Multiple image names should be separated by a plus (+) sign.
+"""
+import time
+import argparse
+import json
+import logging
+
+import requests
+
+workspace_internal_url = "http://workspace-token-service"
+logging.basicConfig(level=logging.INFO, format='%(message)s')
+
+def main():
+    args = parse_args()
+    tester = WorkspaceLaunchTest(commons_url=args.commons_url, access_token=args.access_token, images=args.images.split("+")) # Images passed from the kubernetes jobs is separated by a plus sign "+"
+    tester.initialize_workspace_launch_test()
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Test Workspaces Launch"
+    )
+    parser.add_argument(
+        "--commons-url",
+        dest="commons_url",
+        help="Specify the Commons URL to test"
+    )
+    parser.add_argument(
+        "--access-token",
+        dest="access_token",
+        help="User's access token. It should have the 'credentials' scope since the /launch api requires an access token that can use to get an api key.",
+    )
+    parser.add_argument(
+        "--images",
+        dest="images", 
+        help="Type of image to launch for testing."
+    )
+
+    return parser.parse_args()
+
+class WorkspaceLaunchTest:
+    def __init__(self, commons_url, access_token, images=["(Generic, Limited Gen3-licensed) Stata image"]):
+        self.commons_url = commons_url
+        self.workspace_internal_url = workspace_internal_url
+        self.token_expiration = 0
+        self.headers = {}
+        self.start_time = 0
+        self.end_time = 0
+        self.access_token = access_token
+        self.launch_status = "Workspace did not launch. Something went wrong before launch."
+        self.reason_for_failure = None
+        self.status_response = None
+        self.images = images
+        self.update_headers()
+
+
+    def update_headers(self):
+        """Updates the headers with the current access token."""
+        if self.access_token:
+            self.headers = {"Authorization": f"Bearer {self.access_token}"}
+        else:
+            self.headers = {}
+
+    def get_info_for_image(self, image_name, options=None):
+        for option in options:
+            if option["name"] == image_name:
+                return option
+        return None
+            
+
+    def initialize_workspace_launch_test(self):
+        
+        test_image_ids_map = {} # list of tuples containing image name and image ids
+        
+        # Get available workspace options
+        options_url = self.commons_url + "/lw-workspace/options"
+        try:
+            options_response = requests.get(options_url, headers=self.headers)
+            options_response.raise_for_status()
+
+        except requests.exceptions.RequestException as e:
+            error_msg = f"Couldn't get workspace options with error: {e}"
+            logging.error(error_msg)
+            self.reason_for_failure = error_msg
+        
+        options = options_response.json()
```

Comment:
nitpick: it might be better to convert `options` into a dict here with `name: id` pairs, and then create `available_images` and simplfies some logic in `get_info_for_image()`